### PR TITLE
Entry point for geometry creation - iter 1 - resolves #17

### LINKED ISSUE #17: Add entry point for combining DEMs and potentially a base mesh into multi-polygon and write to disk

### PR DESCRIPTION
**Pull request by SorooshMani-NOAA**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -99,7 +99,7 @@ def _check_gcc_version(self):
     cmdclass={
         'install_jigsaw': InstallJigsawCommand,
         },
-    python_requires='>=3.6, <=3.8',
+    python_requires='>=3.8', #due to typing.Literal
```
Comment (jreniel) [inline setup.py:102]: Shapely doesn't work on python > 3.8

Comment (SorooshMani-NOAA) [reply setup.py:102]: OK, then we need to remove typing.Literal or something. I'm using Python 3.8.5 (from apt) and shapely 1.7.1. Before the addition of typing.Literal I was using Python 3.7.5 on conda

Comment (JaimeCalzadaNOAA) [reply setup.py:102]: No problem with removing typing.Literal...
I am testing this now, and will get back with some comments. On the meantime, if you want to add [tox](https://tox.readthedocs.io/en/latest/) so we can test with different versions you are welcome to do so (this is optional).

File: geomesh/__main__.py [hunk 2]
```diff
@@ -240,39 +256,72 @@ def _logger(self):
 
 
 def parse_args():
```
Comment (jreniel) [inline geomesh/__main__.py:258]: I was hoping to start the parser from scratch, the code in the __main__.py is deprecated. But this is not so important right now.

**Comment by JaimeCalzadaNOAA** [comment]
Let's bring the other piece that we talked about into this commit before we merge. Thanks!

File: geomesh/geom/raster.py [hunk 3]
```diff
@@ -51,6 +51,9 @@ def get_multipolygon(  # type: ignore[override]
         the hull of the raster given optional zmin and zmax contraints.
         """
 
+        if zmin is None and zmax is None:
```
Comment (SorooshMani-NOAA) [inline geomesh/geom/raster.py:54]: One question, what if the raster has a z.mask in some spots? In that case we should consider that too, right? In that case we might want to actually iterate over windows, get the mask for that window, and then collect box of all windows if the window mask doesn't have any -1. In the end like before we can do union of all multipolygons some or all of which can be simple boxes

Comment (JaimeCalzadaNOAA) [reply geomesh/geom/raster.py:54]: This is a mask that generate internally, this is not the raster's _actual_ mask. Returning the bbox multipolygon was tested before, like I mentioned, I left that part out by mistake.

Comment (JaimeCalzadaNOAA) [reply geomesh/geom/raster.py:54]: Actually, using this making technique is a cheap trick to do it fast, if we compute the contours directly, the resulting geom will be even more accurate, but it's more costly and I don't think it's worth the computational expense given what we gain. The 1,-1 trick seems to balance comp. expense vs accuracy well in this case.

Comment (SorooshMani-NOAA) [reply geomesh/geom/raster.py:54]: OK, I see. Thanks

**Pull request merged**